cmd/agent/api/agent/agent.go

```diff
@@ -80,13 +81,23 @@ func getHostname(w http.ResponseWriter, r *http.Request) {
 }
 
 func makeFlare(w http.ResponseWriter, r *http.Request) {
+	var profile *flare.Profile
+
+	err := json.NewDecoder(r.Body).Decode(&profile)
+	switch {
+	case err == io.EOF:
+		log.Debugf("No profile, continuing.")
```

Comment:
Would it not be simpler to check the size of `r.Body` first instead of starting a new Json decoder ?

Reply:
It would be, I did it this way due to the reasoning in [this](https://stackoverflow.com/a/32718077) stackoverflow comment. Do you think we can disregard that as it is just internal communication?

Reply:
I was thinking about something like that:

```go
  if r.Body != "" {
	if err := json.NewDecoder(r.Body).Decode(&profile); err != nil {
		// handle error
	}
  }
```

I don't think we need to instantiate and run the JSON decoder if the body is empty.

Reply:
That makes sense to me, didn't think about doing it that way 👍